core/remotes/docker/fetcher.go

```diff
@@ -564,20 +564,22 @@ func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string,
 						_ = body.Close()
 						_ = writers[i].CloseWithError(err)
 						if err != nil && err != io.EOF {
-							cancelCtx()
 							return err
 						}
 						return nil
 					}
-					if copy() != nil {
-						return
+					if err := copy(); err != nil {
+						return err
```

Comment:
I think we should close the rest of writers for all `j` where `j > i` before returning the error. This will ensure the readers do not block.